### PR TITLE
Update regex

### PR DESCRIPTION
**Pull request by kezuk23**
Updated the regex to allow postcodes as a combination of letters and numbers.

### CODE REVIEW

File: src/main/java/server/InputProcessor.java [hunk 1]
```diff
@@ -21,7 +21,8 @@
         "date", "^\\d{2}/\\d{2}/\\d{4}$",
         "letters", "^[A-Za-z]+$",
         "numbers", "^[0-9]+$",
-        "phonenumber", "^0[\\d]{10}$" //must start with a 0 and be followed by 10 numbers (basic version) OR 1?[\\s-]?\\(?(\\d{3})\\)?[\\s-]?\\d{3}[\\s-]?\\d{4} for international formats (advanced)
+        "phonenumber", "^0[\\d]{10}$", //must start with a 0 and be followed by 10 numbers (basic version) OR 1?[\\s-]?\\(?(\\d{3})\\)?[\\s-]?\\d{3}[\\s-]?\\d{4} for international formats (advanced)
+        "postcode", "^[a-zA-Z0-9]*$"
```
Comment (Ecenursahin) [inline src/main/java/server/InputProcessor.java:25]: Nice addition!

File: src/main/java/server/Registration.java [hunk 2]
```diff
@@ -74,11 +73,7 @@ private Address takeAddress(){
 		String region = InputProcessor.takeValidInput("letters", in, out);
 
 		out.println("Please enter your postcode: ");
-		try {
-			postcode = in.readLine(); // TODO: input processor for postcodes / zip codes?
-		} catch(IOException e){
-			e.printStackTrace();
-		}
+		String postcode = InputProcessor.takeValidInput("postcode", in, out);
```
Comment (Ecenursahin) [inline src/main/java/server/Registration.java:76]: Looks great and much simpler!

**Review by Ecenursahin — COMMENTED** [review]
Great changes looks much simpler, Thank you Simon!

**Review by AllenNinous — APPROVED** [review]
Great additions and well done, Simon!

**Pull request merged**